x/dex/keeper/limitorder.go

```diff
@@ -0,0 +1,231 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/NicholasDotSol/duality/x/dex/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type LimitOrderTranche struct {
+	types.LimitOrderTranche
+	priceInToOut sdk.Dec
+	priceOutToIn sdk.Dec
+}
+
+// Tranche domain object constructor
+func NewLimitOrderTranche(tokenIn string, token0 string, pairId string, tickIndex int64, trancheIndex uint64, reservesIn sdk.Int, reservesOut sdk.Int, totalIn sdk.Int, totalOut sdk.Int) (*LimitOrderTranche, error) {
+	// memoize prices
+	var priceOutToIn sdk.Dec
+	var err error
+	if tokenIn == token0 {
+		priceOutToIn, err = CalcPrice1To0(tickIndex)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		priceOutToIn, err = CalcPrice0To1(tickIndex)
+		if err != nil {
+			return nil, err
+		}
+	}
+	priceInToOut := sdk.OneDec().Quo(priceOutToIn)
+
+	return &LimitOrderTranche{
+		types.LimitOrderTranche{
```

Comment:
No need to have the LimitOrderTranche domain object constructor initialize a LImitOrderTranche data transfer object. That should continue to be separate.